browser/src/UI/components/BufferScrollBar.tsx

```diff
@@ -39,11 +44,49 @@ const ScrollBarWindow = styled.div`
     background-color: rgba(200, 200, 200, 0.2);
     border-top: 1px solid rgba(255, 255, 255, 0.1);
     border-bottom: 1px solid rgba(255, 255, 255, 0.1);
+    pointer-events: none;
 `
 
-export class BufferScrollBar extends React.PureComponent<IBufferScrollBarProps, {}> {
+export class BufferScrollBar extends React.PureComponent<
+    IBufferScrollBarProps,
+    IBufferScrollBarState
+> {
     constructor(props: any) {
         super(props)
+        this.state = { scrollBarTop: 0 }
+        // allow scroll events to be added and removed as event handlers
+        this.trackScroll = this.trackScroll.bind(this)
```

Comment:
@jordan-arenstein thanks so much for making the changes 👍 I'm wondering if the event handlers cant be set in componentDidMount and removed in willUnmount since atm they are created and removed all they time over and over as a user scrolls, so I think the above event handlers in the didMount and willMount would probably do the same job but will only create one set of handlers

Reply:
Can you toggle a handler on and off without destroying it?

Reply:
If not, I don't see any alternative: the event handler is created when the scroll begins and is removed when it ends.

Reply:
ahhh 😆  I didn't quite understand what those event listeners were doing are they only supposed to exist whilst a user is scrolling? in which case the mounting and unmounting wouldnt be the right solution 👍

Reply:
They allow the drag to continue outside of the scrollbar - I figured it was too narrow a target, so the scroll continues as the mouse moves anywhere on the screen

Reply:
I'm taking a look at the arrow functions now.

Reply:
Based on [this MDN link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) `this` is not supported in arrow functions, and so I  can't use state within them

Reply:
or do you mean using arrow functions instead of `method.bind(this)`?

Reply:
yes exactly so instead of

```ts
constructor(props: any) {
        super(props)
        this.state = { scrollBarTop: 0 }
        // allow scroll events to be added and removed as event handlers
        this.trackScroll = this.trackScroll.bind(this)
```

you can do away with the constructor and do

```ts
	public state = {
		scrollBarTop: 0
	}
	
	public endScroll = (e: MouseEvent) => {
	// add content here
	// essentially the this binding will be permanently set to this component so the
	// this cannot be reassigned aka it implicitly works the same as a call to .bind(this)
	}
```

Reply:
fantastic! thank you! I'm testing now, will commit soon after 😄